File: app/src/main/java/com/partycalc/viewmodels/PartyViewModel.java
package com.partycalc.viewmodels;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import com.partycalc.database.Participant;
import com.partycalc.database.Party;

import java.util.List;

public class PartyViewModel extends AndroidViewModel {

    private PartyRepository partyRepository;

    private LiveData<List<Party>> partiesList;


    public PartyViewModel(@NonNull Application application) {
        super(application);

        partyRepository = new PartyRepository(application);
        partiesList = partyRepository.getAllParties();
    }

    public LiveData<List<Party>> getParties() {
        return partiesList;
    }

    public void deleteParty(Party party) {
        partyRepository.removeParty(party);
    }

    public void addParty(Party party) {
        partyRepository.addParty(party);
    }

}

File: app/src/main/java/com/partycalc/viewmodels/PartyViewModel.kt
package com.partycalc.viewmodels

import android.app.Application
import android.util.Log
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.partycalc.database.PartiesDatabase
import com.partycalc.database.Party

class PartyViewModel(application: Application) : AndroidViewModel(application) {
    private val partyRepository: PartyRepository = PartyRepository(PartiesDatabase.getInstance(application))
    val parties: LiveData<List<Party>>

    fun deleteParty(party: Party) {
        partyRepository.removeParty(party)
    }

    fun addParty(party: Party) {
        partyRepository.addParty(party)
    }

    override fun onCleared() {
        super.onCleared()
        Log.i("PartyViewModel", "PartyViewModel destroyed!")
    }

    init {
        parties = partyRepository.getAllParties()
    }
}